varcode/maf.py

```diff
@@ -82,38 +98,60 @@ def load_maf_dataframe(path, nrows=None, verbose=False):
                 df[expected] = df[actual]
                 del df[actual]
             else:
-                raise ValueError("Expected column %s but got %s" % (
-                    expected, actual))
+                error_message = (
+                    "Expected column %s but got %s" % (expected, actual))
+                if raise_on_error:
+                    raise ValueError(error_message)
+                else:
+                    logging.warn(error_message)
+
     return df
 
 
-def load_maf(path, optional_cols=[],
-             sort_key=variant_ascending_position_sort_key):
+def load_maf(
+        path,
+        optional_cols=[],
+        sort_key=variant_ascending_position_sort_key,
+        raise_on_error=True):
     """
     Load reference name and Variant objects from MAF filename.
 
     Parameters
     ----------
+    path : str
+        Path of MAF file
+
     optional_cols : list, optional
         A list of MAF columns to include as metadata if they are present in the MAF.
         Does not result in an error if those columns are not present.
 
     sort_key : fn
         Function which maps each element to a sorting criterion.
         Set to None to not to sort the variants.
+
+    raise_on_error : bool
+        Raise an exception upon encountering an error or just log a warning.
     """
     # pylint: disable=no-member
     # pylint gets confused by read_csv inside load_maf_dataframe
-    maf_df = load_maf_dataframe(path)
+    maf_df = load_maf_dataframe(path, raise_on_error=raise_on_error)
 
-    if len(maf_df) == 0:
+    if len(maf_df) == 0 and raise_on_error:
         raise ValueError("Empty MAF file %s" % path)
 
     ensembl_objects = {}
     variants = []
     metadata = {}
     for _, x in maf_df.iterrows():
         contig = x.Chromosome
+        if not contig or isnan(contig):
+            error_message = "Invalid contig name: %s" % (contig,)
+            if raise_on_error:
```

Comment:
Looks like we're not keeping any malformed variants, which I agree with. Maybe these warning messages should clarify that we're skipping the variant; otherwise could be assumed that they're just malformed or something else?